asgiref/typing.py

```diff
@@ -99,6 +100,7 @@ class HTTPResponseStartEvent(TypedDict):
     type: Literal["http.response.start"]
     status: int
     headers: Iterable[Tuple[bytes, bytes]]
+    trailers: bool
```

Comment:
TypedDict is `total` by default, so this would require `trailers` to always be present which it won't be (as we're doing this as a non-major change by making it optional). We need to switch this so that `trailers` is optional, which may need to be done with two inheriting types due to TypedDict not having fine-grained options.

Reply:
`more_body` is `bool` on `HTTPResponseBodyEvent`. What's the difference between those?

Reply:
> We need to switch this so that `trailers` is optional, which may need to be done with two inheriting types due to TypedDict not having fine-grained options.

We can use https://peps.python.org/pep-0655/#specification

Reply:
What both are suggesting was already discussed when the `asgiref.typing` module was implemented. See https://github.com/django/asgiref/pull/221#issuecomment-751387856.

Reply:
It was, but that was "types for the first time" whereas this is modifying existing types - if we land this change, anything that currently (correctly) typechecks as a response no longer will, and that seems bad.

The suggestion in that thread of Unioning two other dict types together would also work, though.

Reply:
> It was, but that was "types for the first time"

What's the difference? `asgiref` is versioned... :thinking:

Reply:
>  if we land this change, anything that currently (correctly) typechecks as a response no longer will, and that seems bad.

Take in consideration that `uvicorn` is the only ASGI server implementation that uses `asgiref.typing`... :man_shrugging: